jf_agent/agent_logging.py

```diff
@@ -72,3 +123,15 @@ def log_and_print(logger, level, msg, exc_info=False):
     print(msg, flush=True)
     if exc_info:
         print(traceback.format_exc())
+
+
+def log_and_print_error_or_warning(logger, level, error_code,msg_args=[], exc_info=False):
+    '''
+    For a failure that should be sent to the logger, and also written
```

Comment:
It'd be beneficial to have some sort of `assert` in both `log_and_print` and `log_and_print_error_or_warning` to validate that the `level` passed in matches our policy -- i.e., that `log_and_print` is always passed a severity level less than warning, and `log_and_print_error_or_warning` is passed only a severity level >= warning.

I believe you could also take off `exc_info=False` from the former method, since it'd be a programmer's error to call it with a traceback.